lambdas/thumbnail/src/t4_lambda_thumbnail/__init__.py

```diff
@@ -368,8 +368,38 @@ def lambda_handler(request):
         img = format_aicsimage_to_prepped(img)
         # Send to Image object for thumbnail generation and saving to bytes
         img = Image.fromarray(img)
+
         # Generate thumbnail
-        img.thumbnail(size)
+        try:
+            # 'I;16' modes have limited resamplers for scaling (poor quality)
+            # convert to 'I' to use a better resampler.
+            # remove the following two lines to prefer fast (but much lower quality) conversion
+            if img.mode.startswith('I;16'):
+                img = img.convert('I')  # 32-bit (more completely implemented by PIL)
+            # attempt to use the default resampler - we have test images using this.
+            img.thumbnail(size)
+        except ValueError as err:
+            if err.args[0] == 'image has wrong mode':
```

Comment:
```suggestion
            if 'image has wrong mode' in str(err):
```

Reply:
Done.  This change makes the check more general in case they change or add things to the error.